### PR TITLE
Ladspa plugin small memory leak fix

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by jasp00** [comment]
This patch is wrong because you are not supposed to free `const` memory. You should try a different approach.

File: plugins/LadspaEffect/calf/src/plugin.cpp [hunk 1]
```diff
@@ -334,9 +334,13 @@ void ladspa_plugin_metadata_set::prepare(const plugin_metadata_iface *md, LADSPA
     param_count = md->get_param_count(); // XXXKF ladspa_instance<Module>::real_param_count();
     
     const ladspa_plugin_info &plugin_info = md->get_plugin_info();
+	std::string strName(plugin_info.name);
```
Comment (jasp00) [inline plugins/LadspaEffect/calf/src/plugin.cpp]: You should use spaces in this file.

File: plugins/LadspaEffect/calf/src/plugin.cpp [hunk 2]
```diff
@@ -334,9 +334,13 @@ void ladspa_plugin_metadata_set::prepare(const plugin_metadata_iface *md, LADSPA
...
+	strName += " LAPSDA";
+	char *Name = new char[strName.size() + 1];
+	strcpy(Name, strName.c_str());
     descriptor.UniqueID = plugin_info.unique_id;
     descriptor.Label = plugin_info.label;
-    descriptor.Name = strdup((std::string(plugin_info.name) + " LADSPA").c_str());
```
Comment (jasp00) [inline plugins/LadspaEffect/calf/src/plugin.cpp]: `strdup` is fine. You just need to save the result somewhere else. It is not as efficient, but correctness is more important.

File: plugins/LadspaEffect/calf/src/plugin.cpp [hunk 3]
```diff
@@ -467,13 +475,16 @@ ladspa_plugin_metadata_set::~ladspa_plugin_metadata_set()
     delete []descriptor.PortNames;
     delete []descriptor.PortDescriptors;
     delete []descriptor.PortRangeHints;
+	delete []descriptor.Name;
```
Comment (jasp00) [inline plugins/LadspaEffect/calf/src/plugin.cpp]: You are not supposed to delete `const` memory.

File: plugins/LadspaEffect/calf/src/plugin.cpp [hunk 4]
```diff
@@ -467,14 +469,17 @@ ladspa_plugin_metadata_set::~ladspa_plugin_metadata_set()
...
 #if USE_DSSI
     if (presets)
         presets->clear();
     if (preset_descs)
         preset_descs->clear();
     delete presets;
     delete preset_descs;
+    free(descriptorForDssiName);
 #endif
+    delete metadata;
```
Comment (jasp00) [inline plugins/LadspaEffect/calf/src/plugin.cpp]: `metadata` is `const`. You should go to the point where `metadata` is created, make sure you keep a reference, and release it on destruction; you may need a function with `__attribute__((destructor))`.

**Comment by tresf** [comment]
@devnexen I'm sorry this took so long to review, the original reviewer is no longer with the project.  Calf has been moved to a [submodule](https://github.com/lmms/veal) since #3940 and I missed this patch.

Currently @JohannesLorenz is working on getting Calf updated to 0.90, so this patch could have a home in either the current 0.18~ish version on the `ladspa` branch or the 0.90 version on the `master` branch.

Closing since the code this PR is against is no longer part of the LMMS codebase.

**Pull request closed without merging**